flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -221,16 +221,10 @@ private void prepareInflightDataSnapshot(long checkpointId) throws IOException {
 		channelStateWriter.finishOutput(checkpointId);
 	}
 
-	private void finishAndReportAsync(Map<OperatorID, OperatorSnapshotFutures> snapshotFutures, CheckpointMetaData metadata, CheckpointMetrics metrics) {
-		final Future<?> channelWrittenFuture;
-		if (unalignedCheckpointEnabled) {
-			ChannelStateWriteResult writeResult = channelStateWriter.getWriteResult(metadata.getCheckpointId());
-			channelWrittenFuture = CompletableFuture.allOf(
-				writeResult.getInputChannelStateHandles(),
-				writeResult.getResultSubpartitionStateHandles());
-		} else {
-			channelWrittenFuture = FutureUtils.completedVoidFuture();
-		}
+	private void finishAndReportAsync(Map<OperatorID, OperatorSnapshotFutures> snapshotFutures, CheckpointMetaData metadata, CheckpointMetrics metrics, CheckpointOptions options) {
+		Future<?> channelWrittenFuture = unalignedCheckpointEnabled && !options.getCheckpointType().isSavepoint() ?
```

Comment:
Same on ternary.

Reply:
Reverted to minimize changes in this PR.